src/libraries/System.Private.CoreLib/src/System/ArgumentException.cs

```diff
@@ -118,11 +118,31 @@ public static void ThrowIfNullOrEmpty([NotNull] string? argument, [CallerArgumen
             }
         }
 
+        /// <summary>Throws an exception if <paramref name="argument"/> is null, empty, or consists only of white-space characters.</summary>
+        /// <param name="argument">The string argument to validate.</param>
+        /// <param name="paramName">The name of the parameter with which <paramref name="argument"/> corresponds.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="argument"/> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="argument"/> is empty or consists only of white-space characters.</exception>
+        public static void ThrowIfNullOrWhiteSpace([NotNull] string? argument, [CallerArgumentExpression(nameof(argument))] string? paramName = null)
+        {
+            if (string.IsNullOrWhiteSpace(argument))
+            {
+                ThrowNullOrWhiteSpaceException(argument, paramName);
+            }
+        }
+
         [DoesNotReturn]
         private static void ThrowNullOrEmptyException(string? argument, string? paramName)
         {
             ArgumentNullException.ThrowIfNull(argument, paramName);
             throw new ArgumentException(SR.Argument_EmptyString, paramName);
         }
+
+        [DoesNotReturn]
+        private static void ThrowNullOrWhiteSpaceException(string? argument, string? paramName)
+        {
+            ArgumentNullException.ThrowIfNull(argument, paramName);
```

Comment:
Just trying to learn: I remember a comment from @AndyAyersMS a while back that the JIT knows how to recognize throw helpers only if they have a single basic block. Is that no longer the case? As in, given this method has a branch and a throw, and then another throw after that, it seems it doesn't have a single basic block, so would it still be recognized correctly as a throw helper so that the codegen in the caller is the one we expect? 🤔

Reply:
I've just checked and it's recognized as expected. Afair, it's not limitted to a single basic block. Instead, it counts "exits" (returns) and if the number of them is 0 (no returns) -> it's a throw helper

Reply:
Oh, cool, TIL. Thank you! 😄